maestro-orchestra-models/src/main/java/maestro/orchestra/MaestroCommand.kt

```diff
@@ -19,88 +19,50 @@
 
 package maestro.orchestra
 
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.JsonSerializer
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+
+/**
+ * The Mobile.dev platform uses this class in the backend and hence the custom
+ * serialization logic. The earlier implementation of this class had a nullable field for
+ * each command. Sometime in the future we may move this serialization logic to the backend
+ * itself, where it would be more relevant.
+ */
+@JsonSerialize(using = MaestroCommandSerializer::class)
 data class MaestroCommand(
-    val tapOnElement: TapOnElementCommand? = null,
-    val tapOnPoint: TapOnPointCommand? = null,
-    val scrollCommand: ScrollCommand? = null,
-    val swipeCommand: SwipeCommand? = null,
-    val backPressCommand: BackPressCommand? = null,
-    val assertCommand: AssertCommand? = null,
-    val inputTextCommand: InputTextCommand? = null,
-    val launchAppCommand: LaunchAppCommand? = null,
-    val applyConfigurationCommand: ApplyConfigurationCommand? = null,
-    val openLinkCommand: OpenLinkCommand? = null,
-    val pressKeyCommand: PressKeyCommand? = null,
-    val eraseTextCommand: EraseTextCommand? = null,
+    val command: Command?,
```

Comment:
Perhaps we can make `Command` a `sealed class` now?

Reply:
We could. I have left a couple of comments related to this change as well.